### DISCUSSION TITLE
influence of "diffeomorphic warp" on predicted moved images

### DISCUSSION THREAD

**Discussion by keepgooooing** [voxelmorph/voxelmorph#375] [Q&A]
Hello,
I am experimenting on 3D ultrasound image registration with voxelmorph (dev branch, Pytorch version).
When I set int_steps=7 as default, the predicted moved images would have many unreal texture and distortion. 
But when int_steps=0, which means non-diffeomorphic warp, the distortion seems to disappear. 
The hyperparameters except "int_steps" are same between two programs. The loss is MSEloss and regularization weight is 0.01.

I am wondering if this phenomenon is caused by the diffeomorphic warp. Or could it be related to the speckle noise in 3D ultrasound?
Here are the screenshot of 3D ultrasound images:
moved-with diffeomorphic warp;
![mse-diff](https://user-images.githubusercontent.com/95667115/144964870-31eff365-6bf3-46a9-b3a2-62f24fb14de9.png)
moved-without diffeomorphic warp;
![mse-nodiff](https://user-images.githubusercontent.com/95667115/144965014-97896987-98de-40a4-8ace-87d9923b0f54.png)
fixed;
![fixed](https://user-images.githubusercontent.com/95667115/144965059-a93064a9-8279-4406-b199-e6cfba232b1c.png)
moving;
![moving](https://user-images.githubusercontent.com/95667115/144965073-3ee0e522-4387-4f7b-a669-b04ba2881413.png)
